src/main/java/au/org/emii/ncdfgenerator/AttributeValueParser.java

```diff
@@ -0,0 +1,202 @@
+
+package au.org.emii.ncdfgenerator;
+
+
+import java.util.List;
+import java.util.ArrayList;
+
+import ucar.ma2.Array;
+import ucar.ma2.DataType;
+
+import au.org.emii.ncdfgenerator.NcdfGeneratorException ;
+
+
+class AttributeValueParser implements IAttributeValueParser
+{
+	// follows ncdf string conventions
+
+	private int skipWhite(String s, int pos)
+	{
+		while( Character.isSpaceChar(peekChar(s , pos) ))
+			++pos;
+		return pos;
+	}
+
+	private int peekChar(String s, int pos)
+	{
+		if(pos < s.length() )
+			return s.charAt( pos );
+		return -1;
+	}
+
+	public AttributeValue parse( String s ) throws NcdfGeneratorException
+	{
+		int pos = 0;
+
+		AttributeValue a = parseAttributes( s, pos ); 
+		if( a != null) {
+			// ensure we got to the end, with nothing trailing
+			pos = a.pos;
+			pos = skipWhite( s, pos);
+			if( pos == s.length() )
+				return a;
+		}
+
+		throw new NcdfGeneratorException( "Couldn't parse attribute value '" + s + "'" );
+	}
+
+	
+	protected AttributeValue parseAttributes( String s, int pos ) throws NcdfGeneratorException
+	{
+		// try to parse as an attribute array, fallback to creating a scalar 
+
+		List<AttributeValue> items = new ArrayList<AttributeValue> (); 
+		AttributeValue a = null;
+		do { 
+			a = parseAttributeValue1( s, pos ); 
+			if( a != null) { 
+				items.add( a) ; 
+				pos = a.pos;
+				pos = skipWhite( s, pos ); 
+			}
+		}
+		while( a != null ); 
+
+		if( items.size() == 1 ) { 
+			return items.get( 0); 
+		}
+		else if ( items.size() > 1 ) {
+			// handle single dimension only at this point...
+			int shape [] = { items.size() };
```

Comment:
I prefer 'shape', since I'm trying to disambiguate it from the concept of a 'dimension' in netcdf. 

Other netcdf code follows thissame convention eg. see  the prototype, 	Array.factory(DataType dataType, int[] shape_) here ->  https://www.unidata.ucar.edu/software/thredds/v4.3/netcdf-java/v4.2/javadoc/ucar/ma2/class-use/DataType.html